### ISSUE TITLE
There was an error while fetching events!

### ISSUE THREAD

**Issue by tobiasschaeferptb** [cdaecke/md_fullcalendar#23]
Hello,

my setup is TYPO3 11.5.28, calendarize 12.4.4 and md_fullcalendar 3.0.2.
If TYPO3 is located in a subdirectory of the root directory (https://www.abc.com/path-to-typo3/) the error "There was an error while fetching events!" is shown in a JS alert box. The reason is that the url https://www.abc.com/index.php?id=1&type=1573738558&storage=39&start=2023-06-26T00%3A00%3A00%2B02%3A00&end=2023-08-07T00%3A00%3A00%2B02%3A00 is used in the ajax request instead of https://www.abc.com/path-to-typo3/index.php?id=1&type=1573738558&storage=39&start=2023-06-26T00%3A00%3A00%2B02%3A00&end=2023-08-07T00%3A00%3A00%2B02%3A00

To fix this problem I modified the file md_fullcalendar/Resources/Public/fullcalendar/lib/main.js (and minimized it again to main.min.js) to find out the correct path to index.php. It works only if config.baseURL is set via Typoscript.
Line 6674:
```
    var eventSourceDef$1 = {
        parseMeta: function (refined) {
            if (refined.url && (refined.format === 'json' || !refined.format)) {
                return {
                    url: refined.url,
                    format: 'json',
                    method: (refined.method || 'GET').toUpperCase(),
                    extraParams: refined.extraParams,
                    startParam: refined.startParam,
                    endParam: refined.endParam,
                    timeZoneParam: refined.timeZoneParam,
                };
            }
            return null;
        },
        fetch: function (arg, success, failure) {
            var meta = arg.eventSource.meta;
            // T. Schaefer: Bug fix for the problem that the loading of the events on websites that are called up via a subdirectory (www.abc.com/path-to-typo3/) failed because the URL www.abc.com/index.php was called.
            // In order for the bug fix to work, a base URL must be set using TypoScript (config.baseURL).
            if (document.getElementsByTagName('base').length > 0) {
                var baseurl = document.getElementsByTagName('base')[0].href;
                if (baseurl.substr(-1) == '/') {
                    baseurl = baseurl.substr(0, baseurl.length - 1);
                }
                // Only add the baseurl in front if meta.url does not start with the string http.
                if (meta.url.indexOf('http') == -1) {
                    meta.url = baseurl + meta.url;
                }
            }
            // Bug fix end

            var requestParams = buildRequestParams$1(meta, arg.range, arg.context);
            requestJson(meta.method, meta.url, requestParams, function (rawEvents, xhr) {
                success({ rawEvents: rawEvents, xhr: xhr });
            }, function (errorMessage, xhr) {
                failure({ message: errorMessage, xhr: xhr });
            });
        },
    };
```

It would be great if the problem could be fixed in a way where config.baseURL isn't necessary.

Best regards,
Tobias Schäfer

**Comment by cdaecke** [comment]
I have just added a fix for your problem.

Please double check the current master, if this is working for you.

Attention:
Be aware, that I have changed the template `Show.html`. So if you have a modified version of this template in place, please adapt the following changes. Use `<f:format.raw>{sitePath}</f:format.raw>index.php` instead of `/index.php` for the ajax url:

https://github.com/cdaecke/md_fullcalendar/blob/ebdfe88a1407d5692a257ec088943df8cdd2c7d3/Resources/Private/Templates/Cal/Show.html#L56

**Comment by tobiasschaeferptb** [comment]
Hi Christoph,

I tested the current master version but it didn't work in my case.
The problem is that this website has 2 subdirectories (https://www.abc.com/sub1/sub2). It looks like your code only recognises the  first subdirectory (sitePath = /sub1/). 

Best regards,
Tobias Schäfer

**Comment by cdaecke** [comment]
Are you sure, that the startpage of your website is reached in `domain.de/sub1/sub2`? 

As far, as I am concerned, the code which I added, should set all subdirectories of the domain:

This is what I use:
https://github.com/cdaecke/md_fullcalendar/blob/ebdfe88a1407d5692a257ec088943df8cdd2c7d3/Classes/Controller/CalController.php#L84

And that is the documentation:
`Path part to frontend, eg. /some/sub/dir/`
https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/RequestLifeCycle/RequestAttributes/NormalizedParams.html#TYPO3\CMS\Core\Http\NormalizedParams::getSitePath

**Comment by tobiasschaeferptb** [comment]
Hi Christoph,

yes, I can confirm that if the site configuration is set to domain.de/sub1/sub2 as entry point the getSitePath function only returns sub1. If you use getRequestUri instead and add a slash at the end it works.

Best regards,
Tobias Schäfer

**Comment by cdaecke** [comment]
@tobiasschaeferptb Could you please try the current master of the extension? This should be fixed now.

**Comment by cdaecke** [comment]
This should be fixed with #25

**Comment by tobiasschaeferptb** [comment]
The new version 4.0.0 works now. Thank you very much.
